### PR TITLE
New trpc route that starts transfer workflow

### PR DESCRIPTION
**Pull request by youngkidwarrior**

### CODE REVIEW

**Comment by youngkidwarrior** [comment]
> [!WARNING]
> <b>This pull request is not mergeable via GitHub because a downstack PR is open. Once all requirements are satisfied, merge this PR as a stack <a href="https://app.graphite.dev/github/pr/0xsend/sendapp/687?utm_source=stack-comment-downstack-mergeability-warning" >on Graphite</a>.</b>
> <a href="https://graphite.dev/docs/merge-pull-requests">Learn more</a>

* **#688** <a href="https://app.graphite.dev/github/pr/0xsend/sendapp/688?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#687** <a href="https://app.graphite.dev/github/pr/0xsend/sendapp/687?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* **#686** <a href="https://app.graphite.dev/github/pr/0xsend/sendapp/686?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* `dev`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @youngkidwarrior and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

File: packages/api/src/routers/transfer.ts [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+    .mutation(async ({ input }) => {
+      const { sender, nonce } = input
+      try {
+        const connection = await Connection.connect()
+        const client = new Client({
+          connection,
+        })
+
+        const handle = await client.workflow.start(TransferWorkflow, {
+          taskQueue: 'monorepo',
```
Comment (0xBigBoss) [inline packages/api/src/routers/transfer.ts:31]: let's see if we can create some shared constants or configs like we do with `import { config } from @my/wagmi`

File: packages/api/src/routers/transfer.ts [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+          workflowId: `transfer-workflow-${sender}-${nonce}`,
```
Comment (0xBigBoss) [inline packages/api/src/routers/transfer.ts:32]: same here, we should see if we can share the same workflow ID generation code.

File: packages/api/package.json [hunk 3]
```diff
@@ -15,6 +16,7 @@
     "@wagmi/core": "^2.13.1",
     "app": "workspace:*",
     "debug": "^4.3.6",
+    "json-bigint-patch": "^0.0.8",
```
Comment (0xBigBoss) [inline packages/api/package.json:19]: i didn't see where u imported this.

File: packages/api/src/routers/transfer.ts [hunk 4]
```diff
@@ -0,0 +1,48 @@
...
+          args: [input],
+        })
+        log('Started transfer handle', handle.workflowId)
+        // optional: wait for client result
+        const result = await handle.result()
```
Comment (0xBigBoss) [inline packages/api/src/routers/transfer.ts:37]: yeah we should jsut return an id or something for the UI to continue listening for updates. 

ideally, we can create a separate table in supabase or some other way to allow users to find all of there past temporal workflows.

File: packages/api/src/routers/transfer.ts [hunk 5]
```diff
@@ -0,0 +1,48 @@
...
+        token: z.string().optional(),
+        amount: z.string(),
+        nonce: z.string(),
+        signature: z.string(),
+      })
+    )
+    .mutation(async ({ input }) => {
+      const { sender, nonce } = input
+      try {
+        const connection = await Connection.connect()
```
Comment (0xBigBoss) [inline packages/api/src/routers/transfer.ts:25]: same here, should move to a shared temporal package or config that we can customize with the environment etc.

Comment (0xBigBoss) [reply packages/api/src/routers/transfer.ts:25]: also, not sure about all of the options available to the temporal connection client, but we should keep the connection open and re-create it on every single request.

**Pull request closed without merging**